ui/raidboss/data/07-dt/hunts/shaaloani.ts

```diff
@@ -434,6 +562,233 @@ const triggerSet: TriggerSet<Data> = {
+          awaySide = output.rear!();
+          rotation = 2;
+        } else if (matches.id === '91C3') { // right cleave
+          awaySide = output.rightFlank!();
+          rotation = 1;
+        } else if (matches.id === '91C4') { // left cleave
+          awaySide = output.leftFlank!();
+          rotation = 3;
+        }
+
+        if (pattern) {
+          const safeSpot = sandspoutPattern & pattern;
+          let backFrontSpot: TtokSafeSpots = 0b111_111_111_111;
+          if (safeSpot & TtokSafeSpots.Back) {
+            backFrontSpot = TtokSafeSpots.Back;
+            dir1 = 'back';
+          } else if (safeSpot & TtokSafeSpots.Front) {
+            backFrontSpot = TtokSafeSpots.Front;
+            dir1 = 'front';
+          }
+
+          if (safeSpot & backFrontSpot & TtokSafeSpots.Right) {
+            dir2 = 'right';
+          } else if (safeSpot & backFrontSpot & TtokSafeSpots.Left) {
+            dir2 = 'left';
+          }
+          data.ttokSandOrbOnSet++;
+          data.ttokRotated = rotation;
+          return {
+            alertText: output.triple!({
+              inOut: output.outOfHitbox!(),
+              dir2: output[dir1]!(),
+              dir3: output[dir2]!(),
+            }),
+          };
+        }
+        return {
+          infoText: output.awayFrom!({ out: output.outOfHitbox!(), dir: awaySide }),
+        };
+      },
+    },
+    // The boss does either in; out; in-right + out-left; or out-right + in-left
+    // can be combined with sand orbs explosions.
+    {
+      id: 'Hunt Ttokrrone Desert Tempest',
+      type: 'StartsUsing',
+      netRegex: { id: ttokrroneDesertTempestIds, source: 'Ttokrrone', capture: true },
+      durationSeconds: 7,
+      alertText: (data, matches, output) => {
+        const tempest = ttokrroneDesertTempest[matches.id]!;
+        const pattern = data.ttokSandOrbPatterns[data.ttokSandOrbOnSet];
+        let inOut: TtokrroneDodge = 'unknown';
+        let dir2: TtokrroneDodge | null = null;
+        let dir3: TtokrroneDodge | null = null;
+        let backFrontSpot: TtokSafeSpots = 0b111_111_111_111;
+
+        if (tempest & TtokSafeSpots.In) {
+          inOut = 'in';
+        } else if (tempest & TtokSafeSpots.Out) {
+          inOut = 'out';
+        }
+        if (tempest === TtokSafeSpots.InRight) {
+          dir2 = 'right';
+        } else if (tempest === TtokSafeSpots.InLeft) {
+          dir2 = 'left';
+        }
+
+        if (pattern && data.ttokRotated === 0) {
+          const safeSpot = tempest & pattern;
+
+          if (safeSpot & TtokSafeSpots.Back) {
+            backFrontSpot = TtokSafeSpots.Back;
+            dir3 = 'back';
+          } else if (safeSpot & TtokSafeSpots.Front) {
+            backFrontSpot = TtokSafeSpots.Front;
+            dir3 = 'front';
+          }
+          // if the tempest was only in or out, but the sand pattern needs right or left
+          if (!dir2 && !(safeSpot & 0b010_010_010_010)) {
```

Comment:
```suggestion
          if (dir2 && !(safeSpot & TtokSafeSpots.Middle)) {
```

Reply:
This negation is needed, it covers in/out tempest with a sand orb pattern needing right/left. I changed it to `== null` to be clear though.

Reply:
Yep sorry!  I think I was mid-thought when I started removing the negation to replace it with an explicit check, and then forgot to finish the thought -- so yes, your way is much better. 😄